modules/artifact/src/main/java/org/mule/runtime/module/artifact/api/classloader/RegionClassLoader.java

```diff
@@ -216,19 +232,73 @@ public boolean removeClassLoader(ArtifactClassLoader artifactClassLoader) {
     String normalizedName = normalize(name, true);
+    // Check exported resources and all matching artifacts
     final List<ArtifactClassLoader> artifactClassLoaders = resourceMapping.get(normalizedName);
     if (artifactClassLoaders != null) {
       for (ArtifactClassLoader artifactClassLoader : artifactClassLoaders) {
-        resource = artifactClassLoader.findResource(normalizedName);
-        if (resource != null) {
-          break;
+        URL url = artifactClassLoader.findResource(normalizedName);
+        if (url != null) {
+          return url;
+        }
+      }
+    } else if (name.startsWith(RESOURCE_PREFIX)) {
+      Matcher matcher = GAV_PATTERN.matcher(name);
+      // Check for specific artifact requests
+      if (matcher.matches()) {
+        String groupId = matcher.group(1);
+        String artifactId = matcher.group(2);
+        String version = matcher.group(3);
+        String resource = matcher.group(4);
+        LOGGER.debug("Region request for '{}' in group '{}', artifact '{}' and version '{}'.", resource, groupId, artifactId,
+                     version);
+        String normalizedResource = normalize(resource, true);
+
+        // Check whether it's an exported resource from a matching artifact
+        List<ArtifactClassLoader> exportingArtifactClassLoaders = resourceMapping.get(normalizedResource);
+        if (exportingArtifactClassLoaders != null) {
+          for (ArtifactClassLoader artifactClassLoader : exportingArtifactClassLoaders) {
+            BundleDescriptor descriptor = artifactClassLoader.getArtifactDescriptor().getBundleDescriptor();
+            if (isRequestedArtifact(descriptor, groupId, artifactId, version, () -> {
+              LOGGER.warn("Required version '{}' for artifact '{}:{}' not found. Searching in available version '{}'...",
+                          version, descriptor.getGroupId(), descriptor.getArtifactId(), descriptor.getVersion());
+              return true;
+            })) {
+              return artifactClassLoader.findResource(normalizedResource);
+            }
+          }
+        }
+
+        // Check whether it's a resource from an API dependency, since all those should be considered exported
+        ClassLoaderModel classLoaderModel = this.getArtifactDescriptor().getClassLoaderModel();
+        for (BundleDependency dependency : classLoaderModel.getDependencies()) {
+          Optional<String> classifier = dependency.getDescriptor().getClassifier();
+          if (classifier.isPresent() && API_CLASSIFIERS.contains(classifier.get())) {
+            BundleDescriptor descriptor = dependency.getDescriptor();
+            if (isRequestedArtifact(descriptor, groupId, artifactId, version, () -> false)) {
+              return descriptorMapping.computeIfAbsent(descriptor, (CheckedFunction<BundleDescriptor, URLClassLoader>) d -> {
+                try {
+                  return new URLClassLoader(new URL[] {dependency.getBundleUri().toURL()});
```

Comment:
I would do a close of those classloaders when the region is disposed.

Reply:
You're right, done.